### ISSUE TITLE
Apps v2 - (pre5) `fly deploy -a` with no local `fly.toml` wipes Machine config

### ISSUE THREAD

**Issue by catflydotio** [superfly/flyctl#1638]
I may be getting behind the latest fixes because I'm using the prerelease. Just want to remember what to test later.

To reproduce, start with a running V2 app.  From a directory with no `fly.toml` present, run `fly deploy -a appname`.  Check the machine's config to find that `fly.toml`'s part of the configuration has been wiped.

```
$ fly m status e784459b655483 -d
...
Config:
{
  "init": {},
  "image": "registry.fly.io/jan25:deployment-01GQV3BNBCNGGZMDHGSRWQN90N",
  "metadata": {
    "fly_platform_version": "v2",
    "fly_release_id": "l8w0PB4QQoOwMTg6ywbeaB70",
    "fly_release_version": "2"
  },
  "restart": {},
  "guest": {
    "cpu_kind": "shared",
    "cpus": 1,
    "memory_mb": 256
  }
}
```

**Comment by alichay** [comment]
I can't reproduce this. It might've been fixed in #1757?

**Comment by catflydotio** [comment]
Tested in v0.0.465, and can no longer reproduce.
